src/settings/PathConfigStorage.cpp

```diff
@@ -27,209 +28,209 @@ std::vector<double> PathConfigStorage::getLineWidthFactorPerExtruder(const Slice
+    , (layer_nr == 0)? mesh.settings.get<Ratio>("material_flow_layer_0") : mesh.settings.get<Ratio>("material_flow")
+    , GCodePathConfig::SpeedDerivatives{mesh.settings.get<Velocity>("speed_roofing"), mesh.settings.get<Velocity>("acceleration_roofing"), mesh.settings.get<Velocity>("jerk_roofing")}
 )
 , ironing_config(
     PrintFeatureType::Skin
-    , mesh.getSettingInMicrons("skin_line_width")
+    , mesh.settings.get<coord_t>("skin_line_width")
     , layer_thickness
-    , (layer_nr == 0)? mesh.getSettingInPercentage("material_flow_layer_0") : mesh.getSettingInPercentage("material_flow")
-    , GCodePathConfig::SpeedDerivatives{mesh.getSettingInMillimetersPerSecond("speed_ironing"), mesh.getSettingInMillimetersPerSecond("acceleration_ironing"), mesh.getSettingInMillimetersPerSecond("jerk_ironing")}
+    , (layer_nr == 0)? mesh.settings.get<Ratio>("material_flow_layer_0") : mesh.settings.get<Ratio>("material_flow")
+    , GCodePathConfig::SpeedDerivatives{mesh.settings.get<Velocity>("speed_ironing"), mesh.settings.get<Velocity>("acceleration_ironing"), mesh.settings.get<Velocity>("jerk_ironing")}
 )
 
 , perimeter_gap_config(createPerimeterGapConfig(mesh, layer_thickness, layer_nr))
 {
     infill_config.reserve(MAX_INFILL_COMBINE);
 
-    double flow = (layer_nr == 0) ? mesh.getSettingInPercentage("material_flow_layer_0") : mesh.getSettingInPercentage("material_flow");
+    Ratio flow = (layer_nr == 0) ? mesh.settings.get<Ratio>("material_flow_layer_0") : mesh.settings.get<Ratio>("material_flow");
 
     // Use the infill extruder's settings for the infill config if specified
-    const int infill_extruder_nr = mesh.getSettingAsIndex("infill_extruder_nr");
+    const int infill_extruder_nr = mesh.settings.get<int>("infill_extruder_nr");
     if (infill_extruder_nr != -1)
     {
-        const ExtruderTrain* infill_extruder_train = mesh.p_slice_data_storage->meshgroup->getExtruderTrain(infill_extruder_nr);
-        flow = (layer_nr == 0) ? infill_extruder_train->getSettingInPercentage("material_flow_layer_0") : infill_extruder_train->getSettingInPercentage("material_flow");
+        const ExtruderTrain& infill_extruder_train = Application::getInstance().current_slice->scene.extruders[infill_extruder_nr];
+        flow = (layer_nr == 0) ? infill_extruder_train.settings.get<Ratio>("material_flow_layer_0") : infill_extruder_train.settings.get<Ratio>("material_flow");
     }
 
     for (int combine_idx = 0; combine_idx < MAX_INFILL_COMBINE; combine_idx++)
     {
         infill_config.emplace_back(
                 PrintFeatureType::Infill
-                , mesh.getSettingInMicrons("infill_line_width") * (combine_idx + 1) * line_width_factor_per_extruder[mesh.getSettingAsExtruderNr("infill_extruder_nr")]
+                , mesh.settings.get<coord_t>("infill_line_width") * (combine_idx + 1) * line_width_factor_per_extruder[mesh.settings.get<ExtruderTrain&>("infill_extruder_nr").extruder_nr]
```

Comment:
Together with the previous comment, I would suggest to do it this way:

```
const ExtruderTrain& infill_extruder_train = mesh.settings.get<ExtruderTrain&>("infill_extruder_nr");
Ratio flow = (layer_nr == 0) ? infill_extruder_train.settings.get<Ratio>("material_flow_layer_0") : infill_extruder_train.settings.get<Ratio>("material_flow");
```

In case `infill_extruder_id` is negative, the get function will fallback to the mesh value for the extruder train. So we can remove the negative check and clean up this part a bit.

Reply:
Fixed temporarily: ed8a5b6ac02ac6b035457b41f0c4aba2514c95fb
This should give the correct behaviour with infill but not the rest (as with before the refactor).